### PR TITLE
Toolbar cleanup and tests

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: src/notebook/notebook/toolbar.ts [hunk 1]
```diff
@@ -77,13 +77,13 @@ class NotebookToolbar extends Widget {
     if (index === -1) {
       layout.addChild(widget);
     } else {
-      layout.insertChild(index, widget);
+      layout.insertChild(index + 1, widget);
```
Comment (jasongrout) [inline src/notebook/notebook/toolbar.ts]: Thanks. I thought I saw this bug when doing the toolbar demo, but hadn't investigated it yet...

File: src/notebook/notebook/toolbar.ts [hunk 2]
```diff
@@ -114,14 +114,15 @@ class ToolbarButton extends Widget {
...
-  constructor(className: string, onClick: () => void, tooltip?: string) {
+  constructor(options: ToolbarButton.IOptions = {}) {
     super();
+    options = options || {};
     this.addClass(TOOLBAR_BUTTON);
-    this.addClass(className);
-    this._onClick = onClick;
-    if (tooltip) {
-      this.node.title = tooltip;
+    this._onClick = options.onClick;
```
Comment (jasongrout) [inline src/notebook/notebook/toolbar.ts]: `options.onClick || () => {return void 0;}` to set a default callback.

Comment (blink1073) [inline src/notebook/notebook/toolbar.ts]: Fixed.

Comment (jasongrout) [inline src/notebook/notebook/toolbar.ts]: yeah, or what you did :).

**Pull request merged**